packages/dmn-js-shared/src/features/modeling/behavior/NameChangeBehavior.js

```diff
@@ -0,0 +1,87 @@
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+import {
+  getBusinessObject,
+  is
+} from 'dmn-js-shared/lib/util/ModelUtil';
+
+
+export default class NameChangeBehavior extends CommandInterceptor {
+
+  static $inject = [ 'eventBus', 'modeling' ];
+
+  constructor(eventBus, modeling) {
+    super(eventBus);
+
+    this._modeling = modeling;
+
+    this.postExecuted('element.updateProperties', this.updateVariableFromElement);
+    this.postExecuted('element.updateLabel', this.updateVariableFromLabel);
+  }
+
+  updateVariableFromLabel = ({ context }) => {
+    const { element, newLabel } = context;
+    const bo = getBusinessObject(element),
+          variable = bo.variable;
+
+    if (!variable) {
+      return;
+    }
+
+    this._modeling.updateProperties(variable, { name: newLabel });
+  };
+
+  updateVariableFromElement = ({ context }) => {
+    const { element, properties } = context;
+    const bo = getBusinessObject(element);
+
+    if (!bo.variable) {
+      return;
+    }
+
+    if (!(is(element, 'dmn:Decision') || is(element, 'dmn:BusinessKnowledgeModel'))) {
+      return;
+    }
+
+    if (!this.isNameChanged(properties)) {
+      return;
+    }
+
+    if (this.isVariable(element)) {
+      return;
+    }
+
+    else if (!this.isElementVariable(element)) {
+      this.syncElementVariableChange(bo);
+    }
+  };
+
+  isNameChanged(properties) {
+    return 'name' in properties;
+  }
+
+  isVariable(element) {
+    const parent = getParent(element);
+    return (
+      is(element, 'dmn:InformationItem') &&
+      parent && parent.get('variable') === element
+    );
+  }
+
+  isElementVariable(element) {
+    const variable = element.get('variable');
```

Comment:
This breaks when using this feature along with the [dmn-js-properties-panel](https://github.com/bpmn-io/dmn-js-properties-panel).

Reply:
Cf. https://github.com/bpmn-io/dmn-js-properties-panel/issues/103

Reply:
Ouch. I can reproduce it on the Modeler nightly too.